firmware/shared/src/app/App_SharedHeartbeatMonitor.c

```diff
@@ -45,7 +36,8 @@ bool App_SharedHeartbeatMonitor_Tick(struct HeartbeatMonitor *const heartbeat_mo
         heartbeat_monitor->previous_timeout_ms = current_ms;
 
         // Check if the board received all the heartbeats it's listening for
-        heartbeat_monitor->status = heartbeat_monitor->heartbeats_to_check == heartbeat_monitor->heartbeats_checked_in;
+        heartbeat_monitor->status =
+            ~(heartbeat_monitor->heartbeats_to_check) | (heartbeat_monitor->heartbeats_checked_in);
```

Comment:
Could add some more documentation? Not super clear how this works just by looking at it.
Also consider keeping track of heartbeat statuses as an array if its more clear - whatever you think is best.